ui/packages/catalog-ui-search/src/main/webapp/js/model/QueryResponse.js

```diff
@@ -122,16 +122,21 @@ module.exports = Backbone.AssociatedModel.extend({
                         handled = true;
                         res.options = options;
                         switch (res.code) {
-                            case 400:
-                            case 404:
-                            case 500:
+                            case -32000:
+                                if (rpc !== null) {
+                                    rpc.close();
+                                    rpc = null;
+                                }
                                 options.error({
-                                    responseJSON: res
+                                    message: 'User not logged in.'
```

Comment:
Are there any more errors the backend can generate?

Reply:
This will only execute when the error code is `-32000`, which is currently being used specifically when the user is not logged in.